modules/mobile/app/models/mobile/v0/adapters/allergy_intolerance.rb

```diff
@@ -0,0 +1,122 @@
+# frozen_string_literal: true
+
+module Mobile
+  module V0
+    module Adapters
+      class AllergyIntolerance
+        def parse(allergies_info)
+          Array.wrap(allergies_info).map do |allergy|
+            allergy_info = allergy['resource']
+            Mobile::V0::AllergyIntolerance.new(
+              id: allergy_info['id'],
+              resourceType: allergy_info['resourceType'],
+              type: allergy_info['type'],
+              clinicalStatus: clinical_status(allergy_info['clinicalStatus']),
+              code: code(allergy_info['code']),
+              recordedDate: allergy_info['recordedDate'],
+              patient: patient(allergy_info['patient']),
+              notes: notes(allergy_info['note']),
+              reactions: reactions(allergy_info['reaction'])
+            )
+          end
+        end
+
+        private
+
+        def clinical_status(attributes)
+          values = Array.wrap(attributes['coding'])
+          coding_hash = values.map do |code|
+            {
+              'system' => code['system'],
+              'code' => code['code']
+            }
+          end
+
+          { 'coding' => coding_hash }
+        end
+
+        def code(attributes)
+          values = Array.wrap(attributes['coding'])
+          coding_hash = values.map do |code|
+            {
+              'system' => code['system'],
+              'code' => code['code'],
+              'display' => code['display']
+            }
+          end
+
+          {
+            'coding' => coding_hash,
+            'text' => attributes['text']
+          }
+        end
+
+        def patient(attributes)
+          {
+            'reference' => attributes.dig('patient', 'reference'),
+            'display' => attributes.dig('patient', 'display')
+          }
+        end
+
+        def recorder(attributes)
```

Comment:
Is this method being used?

Reply:
Ah good catch, it was supposed to be used but I missed it when comparing schemas. Now it's in use and I updated the specs/docs